code/game/objects/machinery/computer/camera_advanced.dm

```diff
@@ -200,11 +200,21 @@
 		update_ai_detect_hud()
 		if(use_static != USE_STATIC_NONE)
 			GLOB.cameranet.visibility(src, GetViewerClient(), null, use_static)
-		if(visible_icon)
-			if(eye_user.client)
-				eye_user.client.images -= user_image
-				user_image = image(icon,loc,icon_state,FLY_LAYER)
-				eye_user.client.images += user_image
+		if(visible_icon && eye_user.client)
+			eye_user.client.images -= user_image
+			var/atom/top
+			for(var/i in loc)
+				var/atom/A = i
+				if(!top)
+					top = loc
+				if(A.type in GLOB.ignored_atoms)
```

Comment:
Will this work for `/mob/dead/observer` for example? If not, how about a typecache?

Reply:
This would have to be multiple typecaches and affect performance negatively, I'll just make it mob/dead/observer instead of mob/dead since thats the only one we have

Reply:
The performance impact would be only in setting it up, though.
The idea is that you make a typecache of every item on the list, merge all of them into a single global list, and then use that for the check. Then performance should be unaffected.
Makes sense?

Reply:
Typecacheof makes a list that has all subtypes of the supplied types unless you specifically tell it not to

Reply:
Yeah looking again just use a typecache in the glob and use is_type_in_typecache, it's faster than this, O(logn)vs O(n)